ns/nns.go

```diff
@@ -25,31 +24,9 @@ import (
 type NNS struct {
 	nnsContract util.Uint160
 
-	neoClient neoClient
-}
-
-// represents virtual connection to Neo network used by NNS.Dial.
-type neoClient interface {
-	// calls specified method of the Neo smart contract with provided parameters.
-	call(contract util.Uint160, method string, prm []smartcontract.Parameter) (*result.Invoke, error)
-}
-
-// implements neoClient using Neo HTTP client.
-//
-// note: see NNS.Dial to realize why this isn't defined as type wrapper like neoWebSocket.
-type neoHTTP struct {
-	*rpcclient.Client
-}
-
-func (x *neoHTTP) call(contract util.Uint160, method string, prm []smartcontract.Parameter) (*result.Invoke, error) {
-	return x.Client.InvokeFunction(contract, method, prm, nil)
-}
-
-// implements neoClient using Neo WebSocket client.
-type neoWebSocket rpcclient.WSClient
-
-func (x *neoWebSocket) call(contract util.Uint160, method string, prm []smartcontract.Parameter) (*result.Invoke, error) {
-	return (*rpcclient.WSClient)(x).InvokeFunction(contract, method, prm, nil)
+	invoker interface {
```

Comment:
In this context it's still a `neoClient` (state var name).

Reply:
It'd be also nice to leave the docs about what is this: `string`, `...interface{}`?? See how it was previously done.